synthesizer/process/src/lib.rs

```diff
@@ -80,7 +79,7 @@ pub struct Process<N: Network> {
     /// The universal SRS.
     universal_srs: Arc<UniversalSRS<N>>,
     /// The mapping of program IDs to stacks.
-    stacks: IndexMap<ProgramID<N>, Arc<Stack<N>>>,
+    stacks: Arc<RwLock<IndexMap<ProgramID<N>, Arc<Stack<N>>>>>,
```

Comment:
do we actually need an `RwLock` here? I only see a single call to `write`, and it's in the context of `&mut self`

Reply:
ah, but the entire collection is now `Arc`ed; I guess that's fine if we expect it to be large